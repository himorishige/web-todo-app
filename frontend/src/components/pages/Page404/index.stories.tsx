/** @jsxImportSource @emotion/react */
import { ComponentStory, ComponentMeta } from '@storybook/react';
import Page404 from '.';

export default {
  title: 'WebToDoApp/Pages/Page404',
  component: Page404,
} as ComponentMeta<typeof Page404>;

const Template: ComponentStory<typeof Page404> = (args) => <Page404 {...args} />;

export const Default = Template.bind({});
Default.storyName = 'Default';
