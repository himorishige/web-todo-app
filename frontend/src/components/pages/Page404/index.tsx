/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';
import { Layout } from 'src/components/templates';

type Props = {};

const Page404: React.VFC<Props> = () => {
  return (
    <Layout>
      <div css={footerStyle}>404</div>
    </Layout>
  );
};

export default Page404;

const footerStyle = css`
  display: flex;
  align-items: center;
  padding: 1rem;
`;
