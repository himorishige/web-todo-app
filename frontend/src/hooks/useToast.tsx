/** @jsxImportSource @emotion/react */
import React from 'react';
import { css, keyframes } from '@emotion/react';

type TOAST_TYPE = 'SUCCESS' | 'FAIL';

const ToastContext = React.createContext({
  showToast: (toastType: TOAST_TYPE, message: string) => {
    console.log(message);
  },
});

type Props = {
  children: React.ReactNode;
};

export const ToastProvider: React.VFC<Props> = (props) => {
  const [state, update] = React.useState({
    show: false,
    message: 'message',
    toastType: 'SUCCESS',
  });

  const showToast = (toastType: TOAST_TYPE, message: string) => {
    update({ show: true, message, toastType });
    setTimeout(() => update({ ...state, show: false, message: message }), 4000);
  };

  return (
    <>
      <ToastContext.Provider value={{ showToast }}>{props.children}</ToastContext.Provider>
      {state.show && (
        <div css={[toastStyle, state.toastType === 'FAIL' && failStyle]}>{state.message}</div>
      )}
    </>
  );
};

export const useToast = () => React.useContext(ToastContext);

const FadeIn = keyframes`
from {
  opacity: 0;
  top: 1rem;
}
to {
  opacity: 1;
  top: 2rem;
}
`;

const FadeOut = keyframes`
from {
  opacity: 1;
  top: 2rem;
}
to {
  opacity: 0;
  top: 1rem;
}
`;

const toastStyle = css`
  animation: ${FadeIn} 0.3s, ${FadeOut} 0.3s 3s forwards;
  position: fixed;
  display: flex;
  align-items: center;
  justify-content: center;
  min-width: 200px;
  min-height: 40px;
  background: var(--primary-color);
  padding: 1rem;
  top: 2rem;
  left: 50%;
  transform: translate(-50%, 0);
  border-radius: 4px;
  color: white;
`;

const failStyle = css`
  background: red;
`;
