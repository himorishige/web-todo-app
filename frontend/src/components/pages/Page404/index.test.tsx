import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { store } from 'src/app/store';
import { HelmetProvider } from 'react-helmet-async';
import Page404 from '.';
import { ToastProvider } from 'src/hooks/useToast';

describe('pages/Page404', () => {
  test('正しくレンダリングされている', () => {
    const target = render(
      <Provider store={store}>
        <HelmetProvider>
          <BrowserRouter>
            <ToastProvider>
              <Page404 />
            </ToastProvider>
          </BrowserRouter>
        </HelmetProvider>
      </Provider>,
    );
    expect(target).toMatchSnapshot();
  });
});
