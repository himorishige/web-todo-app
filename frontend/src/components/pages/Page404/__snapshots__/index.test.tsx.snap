// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Page404 正しくレンダリングされている 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="css-1hrd1c8"
        >
          <div
            class="css-1hh8z22"
          >
            <div>
              <a
                href="/"
              >
                <svg
                  class="css-1443oa1"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M14 1H2a1 1 0 00-1 1v12a1 1 0 001 1h12a1 1 0 001-1V2a1 1 0 00-1-1zM2 0a2 2 0 00-2 2v12a2 2 0 002 2h12a2 2 0 002-2V2a2 2 0 00-2-2H2z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M15 4H1V3h14v1z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M5 15.5v-14h1v14H5zm5 0v-14h1v14h-1z"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M15 8H1V7h14v1zm0 4H1v-1h14v1z"
                    fill-rule="evenodd"
                  />
                  <path
                    d="M0 2a2 2 0 012-2h12a2 2 0 012 2v2H0V2z"
                  />
                </svg>
              </a>
            </div>
            <h1
              class="css-1llq2fx"
            >
              ToDo App
            </h1>
            <div
              class="css-knz8lr"
            >
              <div
                class="css-kknodv"
              >
                <svg
                  class="css-ba990d"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M2.866 14.85c-.078.444.36.791.746.593l4.39-2.256 4.389 2.256c.386.198.824-.149.746-.592l-.83-4.73 3.523-3.356c.329-.314.158-.888-.283-.95l-4.898-.696L8.465.792a.513.513 0 00-.927 0L5.354 5.12l-4.898.696c-.441.062-.612.636-.283.95l3.523 3.356-.83 4.73zm4.905-2.767l-3.686 1.894.694-3.957a.565.565 0 00-.163-.505L1.71 6.745l4.052-.576a.525.525 0 00.393-.288l1.847-3.658 1.846 3.658a.525.525 0 00.393.288l4.052.575-2.906 2.77a.564.564 0 00-.163.506l.694 3.957-3.686-1.894a.503.503 0 00-.461 0z"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <main
          class="css-1yvttm9-Layout"
        >
          <div
            class="css-29olxo"
          >
            <div
              class="css-1sbdpyj"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M8 15A7 7 0 108 1a7 7 0 000 14zm0 1A8 8 0 108 0a8 8 0 000 16z"
                  fill-rule="evenodd"
                />
                <path
                  d="M8.93 6.588l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588z"
                />
                <circle
                  cx="8"
                  cy="4.5"
                  r="1"
                />
              </svg>
              <p>
                404 Page Not Found...
                <br />
                ページが見つかりませんでした。
              </p>
            </div>
          </div>
        </main>
        <footer
          class="css-t18ryb-Layout"
        >
          <form>
            <div
              class="css-1qqqf83"
            >
              <div
                class="css-7d0ne1"
              >
                <div
                  class="css-1ago99h"
                >
                  <input
                    class="css-5x3p9d"
                    maxlength="40"
                    name="taskName"
                    placeholder="新しいタスクを入力してください"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="css-75wbz6"
                >
                  <button
                    class="css-k3l7h4"
                    disabled=""
                  >
                    登録
                  </button>
                </div>
              </div>
            </div>
          </form>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="css-1hrd1c8"
      >
        <div
          class="css-1hh8z22"
        >
          <div>
            <a
              href="/"
            >
              <svg
                class="css-1443oa1"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M14 1H2a1 1 0 00-1 1v12a1 1 0 001 1h12a1 1 0 001-1V2a1 1 0 00-1-1zM2 0a2 2 0 00-2 2v12a2 2 0 002 2h12a2 2 0 002-2V2a2 2 0 00-2-2H2z"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M15 4H1V3h14v1z"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M5 15.5v-14h1v14H5zm5 0v-14h1v14h-1z"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M15 8H1V7h14v1zm0 4H1v-1h14v1z"
                  fill-rule="evenodd"
                />
                <path
                  d="M0 2a2 2 0 012-2h12a2 2 0 012 2v2H0V2z"
                />
              </svg>
            </a>
          </div>
          <h1
            class="css-1llq2fx"
          >
            ToDo App
          </h1>
          <div
            class="css-knz8lr"
          >
            <div
              class="css-kknodv"
            >
              <svg
                class="css-ba990d"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M2.866 14.85c-.078.444.36.791.746.593l4.39-2.256 4.389 2.256c.386.198.824-.149.746-.592l-.83-4.73 3.523-3.356c.329-.314.158-.888-.283-.95l-4.898-.696L8.465.792a.513.513 0 00-.927 0L5.354 5.12l-4.898.696c-.441.062-.612.636-.283.95l3.523 3.356-.83 4.73zm4.905-2.767l-3.686 1.894.694-3.957a.565.565 0 00-.163-.505L1.71 6.745l4.052-.576a.525.525 0 00.393-.288l1.847-3.658 1.846 3.658a.525.525 0 00.393.288l4.052.575-2.906 2.77a.564.564 0 00-.163.506l.694 3.957-3.686-1.894a.503.503 0 00-.461 0z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <main
        class="css-1yvttm9-Layout"
      >
        <div
          class="css-29olxo"
        >
          <div
            class="css-1sbdpyj"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M8 15A7 7 0 108 1a7 7 0 000 14zm0 1A8 8 0 108 0a8 8 0 000 16z"
                fill-rule="evenodd"
              />
              <path
                d="M8.93 6.588l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588z"
              />
              <circle
                cx="8"
                cy="4.5"
                r="1"
              />
            </svg>
            <p>
              404 Page Not Found...
              <br />
              ページが見つかりませんでした。
            </p>
          </div>
        </div>
      </main>
      <footer
        class="css-t18ryb-Layout"
      >
        <form>
          <div
            class="css-1qqqf83"
          >
            <div
              class="css-7d0ne1"
            >
              <div
                class="css-1ago99h"
              >
                <input
                  class="css-5x3p9d"
                  maxlength="40"
                  name="taskName"
                  placeholder="新しいタスクを入力してください"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="css-75wbz6"
              >
                <button
                  class="css-k3l7h4"
                  disabled=""
                >
                  登録
                </button>
              </div>
            </div>
          </div>
        </form>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
